### PR TITLE
Update segmentation refiner model

### PR DESCRIPTION
**Pull request by EthanReid**
Crude Claude-ish code to add the HQSam style head to Kestrel. Will follow up with a cleaner less verbose/more efficient implementation.

Note, the new refiner brings us from 86.9% on RefCOCO-M to 88.2%.
Note: A lot of the changes are from black formatting.
Note: **Do not delete branch, I want to look at the commits for a bit.**

### CODE REVIEW

File: kestrel/moondream/vision.py [hunk 1]
```diff
@@ -64,6 +64,27 @@ def vision_encoder(crops: torch.Tensor, module: nn.Module, config: VisionConfig)
     return x
 
 
+def vision_encoder_multiscale(
```
Comment (EthanReid) [inline kestrel/moondream/vision.py]: Note: VE now can return an early feature and the actual vision encoding. We use the early feature in the mask refinement. This idea was lifted directly from HQSam.

**Comment by vikhyat** [comment]
@codex review
